cloud_blobstore/gs.py

```diff
@@ -153,9 +171,9 @@ def delete(self, bucket: str, key: str):
         Deletes an object in a bucket.  If the operation definitely did not delete anything, return False.  Any other
         return value is treated as something was possibly deleted.
         """
-        bucket_obj = self._ensure_bucket_loaded(bucket)
-        blob_obj = bucket_obj.get_blob(key)
-        if blob_obj is None:
+        try:
+            blob_obj = self._get_blob_obj(bucket, key)
+        except BlobNotFoundError:
             return False
         blob_obj.delete()
```

Comment:
Out of the scope of this diff, but I think this is probably subject to the same race conditions that forced us to wrap a bunch of stuff with:

```
try:
    xxx
except NotFound:
    yyy
```

rationale: get_blob_obj gets a specific generation of a key, such that if someone overwrites that data, then it's possible that even if the key is still present.

it may be worth switching to [this api](https://googlecloudplatform.github.io/google-cloud-python/latest/storage/buckets.html#google.cloud.storage.bucket.Bucket.delete_blob) as it is less network roundtrips, i believe.